File: OJ/LeetCode/src/1614 - Maximum Nesting Depth of the Parentheses [Ergodic].py
class Solution:
    def maxDepth(self, s: str) -> int:
        depth = 0
        top = 0
        for c in s:
            if c == '(':
                top += 1
            elif c == ')':
                top -= 1
            depth = max(top, depth)

        return depth

File: OJ/LeetCode/src/1614 - Maximum Nesting Depth of the Parentheses [Ergodic].cs
using System;

public class Solution
{
    public int MaxDepth(string s)
    {
        int depth = 0, top = 0;
        foreach (var c in s)
        {
            if (c == '(') top++;
            else if (c == ')') top--;
            depth = Math.Max(top, depth);
        }
        return depth;
    }
}